model/task/db.go

```diff
@@ -985,6 +1015,29 @@ const VersionLimit = 50
 // It attempts to return the most recent display name for each build variant to avoid returning duplicates caused by display names changing.
 // It only checks the last 50 versions that ran for a given task name.
 func FindUniqueBuildVariantNamesByTask(projectId string, taskName string, repoOrderNumber int) ([]*BuildVariantTuple, error) {
+	noLookupPipeline := variantByTaskPipeline(projectId, taskName, repoOrderNumber)
+	lookupPipeline := variantByTaskPipelineWithLookup(projectId, taskName, repoOrderNumber)
+	facet := bson.M{
+		"$facet": bson.M{
```

Comment:
I think similar to the other use of facet, we would want to only split for the necessary section (looking up build variant) and then rejoining, similar to the other linked use of facet. This would simplify the code (not having to have these with/without functions) I would think, since the only thing that should be different is the lookup itself.

Reply:
I had done this because the two pipelines are performing their operations on different subsets of the task collection: those with a BV display name set, and those without it set. Therefore the $match stages are different as well. However if there is a way to make this work with two different $match stages where one receives a $lookup and the other does not I'm happy to discuss.

Reply:
I think this setup make sense although it would likely only be needed temporarily. Eventually most tasks will have the build variant display name field and this part of the aggregation wont be used anymore. (We only look back 50 mainline commits for this specific query)

Reply:
Yep that sounds good to me; can we add that ticket as a TODO here as well?

Reply:
Sure, created ticket here as well: [EVG-17040](https://jira.mongodb.org/browse/EVG-17040)